storage/blob.go

```diff
@@ -6,13 +6,42 @@ package storage
 import (
 	"context"
 	"io"
+	"os"
 
 	"github.com/zeebo/errs"
 )
 
 // ErrInvalidBlobRef is returned when an blob reference is invalid
 var ErrInvalidBlobRef = errs.Class("invalid blob ref")
 
+// FormatVersion represents differing storage format version values. Changes in FormatVersion
+// might affect how a Blobs or BlobReader or BlobWriter instance works, or they might only be
+// relevant to some higher layer. A FormatVersion must be specified when writing a new blob,
+// and the blob storage interface must store that value with the blob somehow, so that the same
+// FormatVersion is returned later when reading that stored blob.
+type FormatVersion int
+
+const (
+	// FormatV0 is the identifier for storage format v0, which also corresponds to an absence of
+	// format version information.
+	FormatV0 FormatVersion = 0
+	// FormatV1 is the identifier for storage format v1
+	FormatV1 FormatVersion = 1
```

Comment:
Because of how we do a reverse iteration over these `FormatVersion`s in `storage/filestore/dir.go` I think it would be good to add a comment here about that. If someone, for example, makes a `FormatV99 = 99` next, it could really hurt performance.